src/components/views/messages/MessageActionBar.tsx

```diff
@@ -274,14 +275,19 @@ const ReplyInThreadButton: React.FC<IReplyInThreadButton> = ({ mxEvent }) => {
 
 interface IFavouriteButtonProp {
     mxEvent: MatrixEvent;
+    favouriteMessagesStore?: FavouriteMessagesStore;
 }
 
-const FavouriteButton: React.FC<IFavouriteButtonProp> = ({ mxEvent }) => {
-    const { isFavourite, toggleFavourite } = useFavouriteMessages();
+const FavouriteButton: FC<IFavouriteButtonProp> = ({ mxEvent, favouriteMessagesStore }: IFavouriteButtonProp) => {
+    const { isFavourite, toggleFavourite, registerFavouritesChangedListener } =
+        useFavouriteMessages(favouriteMessagesStore);
+    const [, forceRefresh] = useState([]);
+
+    registerFavouritesChangedListener(() => forceRefresh([]));
```

Comment:
It seems to be a bit hacky here, maybe `useFavouriteMessages` should return something to trigger the rerender ?